### PR TITLE
Add json loads in case literal_eval fails

### PR DESCRIPTION
**Pull request by hod-alpert**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
Added an attempt to json.loads content in which literal_eval failed on.

## Screenshots
image

image

### CODE REVIEW

File: Tests/timestamp_replacer.py [hunk 1]
```diff
@@ -213,8 +213,14 @@ def clean_json_body(self, req: HTTPRequest) -> None:
...
                     content = OrderedDict(literal_eval(content))
                     self.modify_json_body(req, content)
+                    return
                 except Exception:
                     logging.exception(f'failed to run literal_eval on content {content}')
+                try:
+                    content = json.loads(content, object_pairs_hook=OrderedDict)
+                    self.modify_json_body(req, content)
+                except Exception:
+                    logging.exception(f'failed to run json.loads on content {content}')
```
Comment (avidan-H) [inline Tests/timestamp_replacer.py:223]: If you nest this `try`-`except` block under the first exception handler, then there is no need for the `return` statement.

File: Tests/timestamp_replacer.py [hunk 2]
```diff
@@ -213,8 +213,14 @@ def clean_json_body(self, req: HTTPRequest) -> None:
                 try:
                     content = OrderedDict(literal_eval(content))
                     self.modify_json_body(req, content)
+                    return
                 except Exception:
                     logging.exception(f'failed to run literal_eval on content {content}')
+                try:
+                    content = json.loads(content, object_pairs_hook=OrderedDict)
```
Comment (avidan-H) [inline Tests/timestamp_replacer.py:220]: Should we perhaps add a log statement here - something to the effect of `'parsing the request body with "literal_ast" failed - trying with "json.loads"'`? What do you think?

File: Tests/timestamp_replacer.py [hunk 3]
```diff
@@ -327,8 +333,15 @@ def handle_json_body(self, req: HTTPRequest) -> None:
                     content = OrderedDict(literal_eval(content))
                     json_keys = self.determine_problematic_keys(content)
                     self.json_keys.update(json_keys)
+                    return
```
Comment (avidan-H) [inline Tests/timestamp_replacer.py:336]: Same comments as above.

**Review by avidan-H — CHANGES_REQUESTED** [review]

File: Tests/timestamp_replacer.py (hunk 1 above)
Comment (hod-alpert) [reply Tests/timestamp_replacer.py:223]: Correct but that's weird.
I personally don't like nested code
let me know how important it is for you, if it is I'll change it

**Review by avidan-H — APPROVED** [review]

**Pull request merged**